src/util/RoundUtils.cpp

```diff
@@ -53,8 +54,78 @@ namespace spy::util {
 
     void RoundUtils::resetUpdatedMarker(gameplay::State &s) {
         s.getMap().forAllFields([](scenario::Field &field) {
-           field.setUpdated(false);
+            field.setUpdated(false);
         });
     }
+
+    bool RoundUtils::isGameOver(const gameplay::State &s) {
+        // check if cat has diamond collar
+        bool getHasCollar = s.getHasCatDiamondCollar();
+
+        // check if character remaining on field
+        bool charOnField = false;
+        for (const auto &character : s.getCharacters()) {
+            if (character.getCoordinates().has_value()) {
+                charOnField = true;
+                break;
+            }
+        }
+
+        return getHasCollar || !charOnField;
+    }
+
+    void RoundUtils::updateGameOver(gameplay::State &s, const spy::MatchConfig &config) {
+        // convert chips to IP
+        for (auto &character : s.getCharacters()) {
+            character.addIntelligencePoints(character.getChips() * config.getChipsToIpFactor());
+        }
+    }
+
+    spy::character::FactionEnum RoundUtils::determineWinningFaction(const gameplay::State &s) {
+        using character::FactionEnum;
+
+        // check which faction has more ip
+        unsigned int ipFaction1 = 0;
+        unsigned int ipFaction2 = 0;
+
+        for (const auto &character : s.getCharacters()) {
+            if (character.getFaction() == FactionEnum::PLAYER1) {
+                ipFaction1 += character.getIntelligencePoints();
+            } else if (character.getFaction() == FactionEnum::PLAYER2) {
+                ipFaction2 += character.getIntelligencePoints();
+            }
+        }
+
+        if (ipFaction1 != ipFaction2) {
+            return ipFaction1 > ipFaction2 ? FactionEnum::PLAYER1 : FactionEnum::PLAYER2;
+        }
+
+        auto stats = s.getConstFactionStats();
+
+        // which faction brought diamond collar to cat
+        if (stats.collarToCat != FactionEnum::INVALID) {
```

Comment:
Eintrag fehlt in den stats. Dann muss auch executeDiamondCollar angepasst werden